ibis/expr/types/generic.py

```diff
@@ -116,21 +184,26 @@ def cast(self, target_type: dt.DataType) -> Value:
 
         return op.to_expr()
 
-    def try_cast(self, target_type: dt.DataType) -> Value:
+    def try_cast(self, target_type: Any) -> Value:
```

Comment:
IDK if you saw this type hint change @cpcloud. The old one was overly-restrictive. It might be nice if the dtype module exported some type alias for `TypeLike = Union[dt.DataType, str, np.dtype, ...]` so that here and in other places we could consume it.